### PR TITLE
V3

### PR DESCRIPTION
**Pull request by promiseofcake**
Feature release for various improvements:

1. Ability to queue within a given pipeline (have a given workflow wait and run last)
2. Ability to ignore with the queue (now global queue) a given workflow by name when calculating blocking.

### CODE REVIEW

**Comment by promiseofcake** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/promiseofcake/workflow-queue?version=dev:42c06e8548e073464f29c9e8a9bc091e749c86a7

File: .circleci/test-deploy.yml [hunk 1]
```diff
@@ -7,25 +7,51 @@ filters: &filters
...
+      - image: cimg/base:edge
+    parameters:
+      time:
+        type: integer
+    steps:
+      - run: echo "sleeping << parameters.time >>s" &&  sleep << parameters.time >>
+
 workflows:
-  test-deploy:
+  test-pipeline-a:
```
Comment (promiseofcake) [inline .circleci/test-deploy.yml:21]: We expect pipeline b to finish first, c second, a third.

**Comment by promiseofcake** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/promiseofcake/workflow-queue?version=dev:4466452f853360fa4fd2a9ab4462fd31ca8836f3

File: src/jobs/global-queue.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      but not yet active. This number indicates the threshold for API returning no previous pending pipelines.
+      Default is `1` confirmation, increase if you see issues.
+  ignored-workflows:
+    type: string
+    default: ""
+    description: Comma separated list of workflow names to ignore as blocking workflows for the global queue.
+  include-on-hold:
+    type: boolean
+    default: false
+    description: Consider `on-hold`` workflows waiting for approval as running and include them in the queue.
```
Comment (andyhorner-chime) [inline src/jobs/global-queue.yml]:
```suggestion
    description: Consider `on-hold` workflows waiting for approval as running and include them in the queue.
```

File: src/commands/global_block.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      but not yet active. This number indicates the threshold for API returning no previous pending pipelines.
+      Default is `1` confirmation, increase if you see issues.
+  ignored-workflows:
+    type: string
+    default: ""
+    description: Comma separated list of workflow names to ignore as blocking workflows for the global queue.
+  include-on-hold:
+    type: boolean
+    default: false
+    description: Consider `on-hold`` workflows waiting for approval as running and include them in the queue.
```
Comment (andyhorner-chime) [inline src/commands/global_block.yml]:
```suggestion
    description: Consider `on-hold` workflows waiting for approval as running and include them in the queue.
```

**Review by andyhorner-chime — APPROVED** [review]

**Comment by promiseofcake** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/promiseofcake/workflow-queue?version=dev:96876d57282a9062cc4fd004154f853c385291f7

**Comment by promiseofcake** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/promiseofcake/workflow-queue?version=dev:933a202977f532677ff63ab69d622bca28b1ceb1

**Comment by promiseofcake** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/promiseofcake/workflow-queue?version=dev:96876d57282a9062cc4fd004154f853c385291f7

**Comment by promiseofcake** [comment]
Your orb has been published to the CircleCI Orb Registry.
You can view your published orb on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/promiseofcake/workflow-queue?version=3.0.0

**Pull request merged**